src/components/SQFormDialog/SQFormDialogInner.js

```diff
@@ -98,7 +104,7 @@ function SQFormDialogInner({
         maxWidth={maxWidth}
         open={isOpen}
         TransitionComponent={Transition}
-        onClose={handleCancel}
+        onClose={showSecondaryButton ? handleCancel : null}
```

Comment:
I think we want to provide a `noop` instead of `null` here. 
I haven't looked at the MUI `types` for Dialog, but I don't think `onClose` accepts `null`. I know it will accept `undefined` or `() => {}`

`undefined` probably makes the most sense.